### PR TITLE
Adding logger parameters to container-orchestration

### PR DESCRIPTION
**Pull request by GregoryIvo**
Adds the ability to select a logger type & logger parameters when creating a container.
Screen Shot 2022-03-08 at 5 00 44 AM

**Related Issue:** This PR fixes/closes {issue number}

**Description of the solution adopted:** A more detailed description of the changes made to solve/close one or more issues. If the PR is simple and easy to understand this section can be skipped.

**Screenshots:** If applicable, add screenshots to help explain your solution

**Any side note on the changes made:** Description of any other change that has been made, which is not directly linked to the issue resolution [e.g. Code clean up/Sonar issue resolution]

### CODE REVIEW

File: kura/org.eclipse.kura.container.provider/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -5,9 +5,9 @@ Bundle-SymbolicName: org.eclipse.kura.container.provider;singleton:=true
 Bundle-Version: 1.0.0.qualifier
 Bundle-Vendor: EUROTECH
 Bundle-RequiredExecutionEnvironment: JavaSE-1.8
-Import-Package: org.eclipse.kura.container.orchestration.provider;version="[1.0,2.0)",
- org.eclipse.kura;version="[1.3,2.0)",
+Import-Package: org.eclipse.kura;version="[1.3,2.0)",
  org.eclipse.kura.configuration;version="[1.2,1.3)",
+ org.eclipse.kura.container.orchestration.provider,
```
Comment (MMaiero) [inline kura/org.eclipse.kura.container.provider/META-INF/MANIFEST.MF]: range required

File: kura/org.eclipse.kura.container.orchestration.provider/src/main/java/org/eclipse/kura/container/orchestration/provider/impl/DockerServiceImpl.java [hunk 2]
```diff
@@ -657,7 +671,7 @@ public void stopContainer(ContainerDescriptor container) {
             }
 
             if (Boolean.FALSE.equals(container.getIsEsfManaged())) {
-                // container is not Framework Managed, and thus should only be stopped and not deleted.
+                // container is not ESF Managed, and thus should only be stopped and not deleted.
```
Comment (MMaiero) [inline kura/org.eclipse.kura.container.orchestration.provider/src/main/java/org/eclipse/kura/container/orchestration/provider/impl/DockerServiceImpl.java]: ESF Managed -> Framework Managed

File: kura/org.eclipse.kura.container.provider/OSGI-INF/metatype/org.eclipse.kura.container.provider.ConfigurableGenericDockerService.xml [hunk 3]
```diff
@@ -49,6 +49,31 @@
         <AD id="container.device" name="Peripheral Device"
             description="Used to pass physical devices to a container. Example: /dev/gpiomem, /dev/ttyUSB0. (Generally Requires privileged mode to be enabled.)" type="String" cardinality="1"
             required="false" default="" />
+
+        <AD id="container.loggingType" 
+            name="Logger Type"
+            description="Used to pass physical devices to a container. Example: /dev/gpiomem, /dev/ttyUSB0. (Generally Requires privileged mode to be enabled.)" type="String" cardinality="0"
+            required="true" 
+            default="DEFAULT">
+            <Option label="NONE" value="NONE" />
```
Comment (MMaiero) [inline kura/org.eclipse.kura.container.provider/OSGI-INF/metatype/org.eclipse.kura.container.provider.ConfigurableGenericDockerService.xml:58]: Are we sure that all those options work? Do we have corresponding QA tests?

**Comment by MMaiero** [comment]
Can you also check why the CI build is failing with a lot of tests errors?
Check also the coverage: https://sonarcloud.io/summary/new_code?id=org.eclipse.kura%3Akura&branch=PR-3837

File: kura/org.eclipse.kura.container.provider/OSGI-INF/metatype/org.eclipse.kura.container.provider.ConfigurableGenericDockerService.xml (hunk 3 above)
Comment (GregoryIvo) [reply kura/org.eclipse.kura.container.provider/OSGI-INF/metatype/org.eclipse.kura.container.provider.ConfigurableGenericDockerService.xml:58]: These are all the options extended by the the docker-java api, so theoretically they should work.  Will try to figure out a testing methodology with QA team

**Comment by MMaiero** [comment]
It still reports 42 failures in the tests

**Comment by GregoryIvo** [comment]
> It still reports 42 failures in the tests

This is very weird, all tests are passing in my local environment. I think my MVN setup may be broken? Will investigate further.

update: 
image

**Comment by MMaiero** [comment]
> > It still reports 42 failures in the tests
> 
> This is very weird, all tests are passing in my local environment. I think my MVN setup may be broken? Will investigate further.
> 
> update: image

Did you try cleaning up the .m2 folder before running the build and tests?

**Comment by GregoryIvo** [comment]
> > > It still reports 42 failures in the tests
> > 
> > 
> > This is very weird, all tests are passing in my local environment. I think my MVN setup may be broken? Will investigate further.
> > update: image
> 
> Did you try cleaning up the .m2 folder before running the build and tests?

image
Still having the same success after deleting ~/.p2 repo folder, sadly?

**Comment by GregoryIvo** [comment]
@MMaiero  I think it's something to do with the classpath in the pipeline. All the tests appear to be failing because they can't find this one particular class/enum I began using in this PR. still trying to investigate further. 


image

**Comment by code-incubus** [comment]
Hi  @GregoryIvo during checking of this PR, the first thing that I noticed is the same (description) for Peripheral Device and Logger Type labels. ("_Used to pass physical devices to a container. Example: /dev/gpiomem, /dev/ttyUSB0. (Generally Requires privileged mode to be enabled._)")

**Comment by marcellorinaldo** [comment]
Hi @GregoryIvo! I would try to add the following dependencies to the `org.eclipse.kura.container.provider.test/pom.xml`:
```
<dependencies>
    	<dependency>
    		<groupId>com.github.docker-java</groupId>
    		<artifactId>docker-java-api</artifactId>
    		<version>3.2.12</version>
    		<scope>runtime</scope>
    	</dependency>
    	<dependency>
    		<groupId>com.github.docker-java</groupId>
    		<artifactId>docker-java-core</artifactId>
    		<version>3.2.12</version>
    		<scope> runtime </scope>
    	</dependency>
    	<dependency>
    		<groupId>com.github.docker-java</groupId>
    		<artifactId>docker-java-transport</artifactId>
    		<version>3.2.12</version>
    		<scope> runtime </scope>
    	</dependency>
    	<dependency>
    		<groupId>com.github.docker-java</groupId>
    		<artifactId>docker-java-transport-httpclient5</artifactId>
    		<version>3.2.12</version>
    		<scope> runtime </scope>
    	</dependency>
</dependencies>
```

These are the same dependencies of the `org.eclipse.kura.container.orchestration.provider.test/pom.xml`. Transitivity rules might be the problem. https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-scope. I would also try to remove such lines from both, in theory those deps shouldn't be needed in the tests pom and that may cause the errors on the CI.

**Comment by GregoryIvo** [comment]
closed because has been superseded by 3886

**Pull request closed without merging**